src/main/java/graphql/analysis/QueryTraversal.java

```diff
@@ -163,20 +166,31 @@ private GraphQLObjectType getRootTypeFromOperation(OperationDefinition operation
         }
     }
 
-    private List<Node> childrenOf(Node selection) {
-        return childrenOfSelectionProvider.getSelections((Selection) selection);
+    private List<Node> childrenOf(Node<?> node) {
+        if (!(node instanceof FragmentSpread)) {
+            return node.getChildren();
+        }
+        FragmentSpread fragmentSpread = (FragmentSpread) node;
+        return Collections.singletonList(fragmentsByName.get(fragmentSpread.getName()));
     }
 
-    private void visitImpl(QueryVisitor visitFieldCallback, boolean preOrder) {
+    private Object visitImpl(QueryVisitor visitFieldCallback, Boolean preOrder) {
```

Comment:
why use null boolean as a flag.  Surely a boolean is true of false??

Reply:
I need three states, therefore Boolean